### PR TITLE
Use flow-bin to typecheck locally, update ESLint to 2.0

### PR DESCRIPTION
**Pull request by satya164**
Using local version of the Flow make it easy to have project specific version instead of relying on the user to have the correct version installed globally. For example, React Native uses an older version of Flow, while I have the latest version installed. Now I cannot typecheck the code because my version doesn't match the `.flowconfig`.

**Test plan (required)**

Run `npm run lint` and `npm run flow` to run `eslint` and `flow`.

cc @bestander @mkonicek

### CODE REVIEW

**Comment by bestander** [comment]
@facebook-github-bot import

File: package.json [hunk 1]
```diff
@@ -109,8 +112,9 @@
     "npm-shrinkwrap.json"
   ],
   "scripts": {
-    "test": "NODE_ENV=test jest",
-    "lint": "eslint Examples/ Libraries/",
+    "test": "NODE_ENV=test node_modules/jest-cli/bin/jest.js",
```
Comment (cpojer) [inline package.json]: this shouldn't be necessary? `npm install` creates a `.bin` folder in `node_modules` with a `jest` executable. `npm run` adds this to the $PATH.

**Comment by cpojer** [comment]
Can you clarify the "use local tools"? Your explanation doesn't make sense in the context of how npm works. It might make sense for flow, but not for jest or eslint.

File: package.json (hunk 1 above)
Comment (satya164) [inline package.json]: @cpojer My bad. For some reason it didn't work for me last time I tried. Will update the PR.

**Comment by satya164** [comment]
@cpojer Updated!

**Comment by bestander** [comment]
Waiting for @cpojer to approve this.
I'll reimport, because this touches npm deps, I'll need to update internal npm deps with a script.

**Comment by cpojer** [comment]
Yeah I think this makes more sense.

**Comment by mkonicek** [comment]
Thanks for doing this!

Please hold on with merging this until Travis tests are back to green to be safe.

**Comment by bestander** [comment]
Added tag so that we come back to this next week

**Comment by bestander** [comment]
@satya164 maybe we can remove global flow install in Travis then?
https://github.com/facebook/react-native/blob/master/.travis.yml#L18

**Comment by satya164** [comment]
@bestander done!

**Comment by mkonicek** [comment]
e2e tests are still broken so we can't merge this yet, thanks for the patience.

**Comment by bestander** [comment]
@facebook-github-bot import

**Comment by bestander** [comment]
shipping...

**Pull request closed without merging**